src/index.ts

```diff
@@ -685,48 +704,36 @@ export function processProject(
   project.getSourceFiles().forEach(sourceFile => {
     const dependencies: Dependencies = new Map()
     const addDependency = createAddDependency(dependencies)
-    const functions = sourceFile
-      .getChildAtIndex(0)
-      .getChildren()
-      .reduce<string[]>((acc, child) => {
-        if (!TypeGuards.isJSDocableNode(child)) {
-          return acc
-        }
-        const typeGuardName = getTypeGuardName(child.getJsDocs())
-        if (typeGuardName === null) {
-          return acc
-        }
-        if (!TypeGuards.isExportableNode(child)) {
-          reportError(`Must be exportable:\n\n${child.getText()}\n`)
-          return acc
-        }
+
+    const functions = []
+    const exports = Array.from(sourceFile.getExportedDeclarations().values())
+    for (const exp of exports) {
+      for (const singleExport of exp) {
         if (
-          TypeGuards.isEnumDeclaration(child) ||
```

Comment:
This is intended to work with enums, but I just checked and saw there is no test coverage. Unless it's not needed, could you restore this line? Ideally we'd add some test coverage for enums too.

Reply:
Resolved here https://github.com/rhys-vdw/ts-auto-guard/pull/106#discussion_r512332802